project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/c4m_user_notifications.module

```diff
@@ -121,31 +121,40 @@ function c4m_user_notifications_og_membership_update($og_membership) {
     c4m_user_notifications_notify_user_membership_accepted($og_membership);
 
     if ($moderated == C4M_OG_MODERATED_GROUP) {
-      // Check membership entity was an invitation or membership request.
-      // If not invitation, og_membership entity og_membership_request
-      // field is an empty array.
-      if (empty($og_membership->og_membership_request)) {
-        // Notify admins only if it's not an invitation to group.
-        // When user is invited, and approves the request through link
-        // at mail, there's no need to notify admins.
+      // When user is invited, and approves the request through link
+      // at mail, there's no need to notify admins about new member.
+      if (_c4m_user_notifications_is_invite_membership($og_membership)) {
         c4m_user_notifications_moderated_group_notify_admins_membership_accepted($og_membership);
       }
     }
     elseif ($moderated == C4M_OG_OPEN_GROUP) {
-      // Check membership entity was an invitation or membership request.
-      // If not invitation, og_membership entity og_membership_request
-      // field is an empty array.
-      if (empty($og_membership->og_membership_request)) {
-        // Notify admins only if it's not an invitation to group.
-        // When user is invited, and approves the request through link
-        // at mail, there's no need to notify admins.
+      // When user is invited, and approves the request through link
+      // at mail, there's no need to notify admins about new member.
+      if (_c4m_user_notifications_is_invite_membership($og_membership)) {
         c4m_user_notifications_open_group_notify_admins_membership_accepted($og_membership);
       }
     }
   }
 }
 
 /**
+ * Check if user was added as a result of invitation.
+ *
+ * When membership was not created as a result of invitation, at membership
+ * entity, og_membership_request field is either an empty array, or it's
+ * only item is an empty array.
+ *
+ * @param object $og_membership
+ *   The membership object.
+ *
+ * @return bool
+ *   TRUE, if user was added as a result of invitation.
+ */
+function _c4m_user_notifications_is_invite_membership($og_membership) {
+  return !empty($og_membership->og_membership_request) && !empty($og_membership->og_membership_request[0]);
```

Comment:
remove this line and use `c4m_og_get_invited_memberships_ids()`, something like:

```php
return in_array($og_membership->id, c4m_og_get_invited_memberships_ids());
```